ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -338,4 +373,42 @@ private void addStatTask(ASTNode root, Table table, Path oldPartitionLocation, P
       moveTask.addDependentTask(statTask);
     }
   }
+
+  public StringBuilder constructDeleteQuery(Table table, Map<String, String> partitionSpec) throws SemanticException {
+    StringBuilder sb = new StringBuilder().append("delete from ").append(table.getTableName())
```

Comment:
This may have to be unescaped
```
table.getTableName()
```
to support quoted table names with special characters
https://github.com/apache/hive/blob/88bc8269a64d31eee372bf3602933c75283c686b/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java#L15838C20-L15843

Reply:
Added a line to get qualified table name - 
```
String qualifiedTableName = HiveTableName.ofNullable(HiveUtils.unparseIdentifier(table.getTableName(), this.conf),
            HiveUtils.unparseIdentifier(table.getDbName(), this.conf), null).getNotEmptyDbTable();
```